### PR TITLE
refctor(sequencing metrics parser)

### PR DESCRIPTION
**Pull request by ChrOertlin**
# Description

Encapsulate bcl convert metrics parsing logic in preparation to re-wiring logic to new DB models.

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### CODE REVIEW

File: cg/meta/demultiplex/demux_post_processing.py [hunk 1]
```diff
@@ -29,8 +29,10 @@ class DemuxPostProcessingAPI:
 
     def __init__(self, config: CGConfig) -> None:
         self.config: CGConfig = config
-        self.flow_cells_dir: Path = Path(config.illumina_flow_cells_directory)
-        self.demultiplexed_runs_dir: Path = Path(config.illumina_demultiplexed_runs_directory)
+        self.flow_cells_dir: Path = Path(config.run_instruments.illumina.flow_cell_runs_dir)
```
Comment (ChrOertlin) [inline cg/meta/demultiplex/demux_post_processing.py:32]: new config paths that we use for run instruments, this already exists in config and servers

File: cg/services/bcl_convert_metrics_service/parser.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
     QUALITY_METRICS_FILE_NAME,
 )
 from cg.io.controller import ReadFile
+from cg.services.bcl_convert_metrics_service.models import SequencingQualityMetrics, DemuxMetrics
```
Comment (diitaz93) [inline cg/services/bcl_convert_metrics_service/parser.py:17]: change to alphabetical order

**Comment by ChrOertlin** [comment]
# Tests

`cg demultiplex finish flow-cell` completed sucessfully and metrics are added in statusDB

File: cg/services/bcl_convert_metrics_service/bcl_convert_metrics_service.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            # BclConvert tags undetermined reads in a lane with the sample id "Undetermined".
+            metrics: SampleLaneSequencingMetrics = self._create_bcl_convert_sequencing_metrics(
+                sample_internal_id=UNDETERMINED, lane=lane, metrics_parser=metrics_parser
+            )
+            metrics.sample_internal_id = sample_internal_id
+            undetermined_metrics.append(metrics)
+        return undetermined_metrics
+
+    @staticmethod
+    def _create_bcl_convert_sequencing_metrics(
```
Comment (diitaz93) [inline cg/services/bcl_convert_metrics_service/bcl_convert_metrics_service.py:56]: I think the bcl_convert in this name is something that I forgot to clean. I think it is not necessary as every Illumina flow cell is demultiplexed using BCLConvert

**Review by diitaz93 — APPROVED** [review]
Great refactoring 💯 I still think that we can get rid of the name BCLConvert in the file name, the class name and in one of the functions. Everything that is Illumina is demultiplexed with BCLConvert, so keeping the name seems redundant to me, maybe this could be explained in the docstring of the class. But it is a minor thing that could be discussed after merging this PR.

File: cg/services/bcl_convert_metrics_service/bcl_convert_metrics_service.py (hunk 3 above)
Comment (ChrOertlin) [reply cg/services/bcl_convert_metrics_service/bcl_convert_metrics_service.py:56]: we can fix it later, these function perhaps will be gone anyway!

**Pull request merged**